pkg/register/restResolver.go

```diff
@@ -91,40 +141,43 @@ func (c *RestResolverClient) GetDocument(ctx context.Context, documentID string)
 
 		neterr, ok := err.(net.Error)
 		if ok && neterr.Timeout() {
-			totalResolverErrors.WithLabelValues(MetricErrorTypeTimeout).Inc()
+			c.meter.RecordError(MetricErrorTypeTimeout)
 		} else {
-			totalResolverErrors.WithLabelValues(MetricErrorTypeConnection).Inc()
+			c.meter.RecordError(MetricErrorTypeConnection)
 		}
 		return nil, &ResolverError{err: err, errType: ConnectionError}
 	}
-	defer response.Body.Close()
+	defer func() {
+		response.Body.Close()
+		c.meter.RecordConnectionReleased()
```

Comment:
Because there is at least one return statement before this, the connection counter is not always "decremented" unless I misread this.